Serve same 404 for assets

AndrewGorton opened the pull request:
It's a bit ugly really, as I'm duplicating the Thymeleaf artifacts which we already create in ThymeleafView/Resolver/Renderer. 

An alternative approach would be to [use CloudFront to mask 404s](http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/custom-error-pages.html).

Ideally you'd want to have the Thymeleaf artifacts injected into the object, but the HK2 ServiceLocator is null at the point of Application.run() and the jetty.setErrorHandler takes a class instance, not a class.

AndrewGorton committed:
Initial ugly attempt to use Thymeleaf in the Jetty custom error handler.

AndrewGorton committed:
Refactored to only create Thymeleaf engine on construction.

AndrewGorton committed:
Merge branch 'master' of github.com:openregister/presentation into serve-same-404-for-assets

AndrewGorton committed:
Added in the register's name to the error message.

philandstuff commented on src/main/java/uk/gov/register/presentation/AssetsBundleCustomErrorHandler.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    @Override
+    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
```
which errors does this handle? is it just 404s or all 4xxs or all 4xxs + all 5xxs?

AndrewGorton replied:
Originally, it handled everything in AssetsBundle. Now I've added a check at the top which ensures it's only for 404s.

AndrewGorton committed:
Only handle 404 errors.

philandstuff commented:
looks like there's an encoding issue -- see "This is a new service ? your feedback"

<img width="579" alt="screen shot 2016-03-17 at 14 51 53" src="https://cloud.githubusercontent.com/assets/581269/13849922/f5ddc5de-ec4f-11e5-8c3f-2ce1920f62c0.png">

Do you want to take a look at it before merging or shall I merge as is and you can look after?

AndrewGorton commented:
Lemme have a looksie. I was checking the /assets123 problem but that's mysteriously vanished instead.

AndrewGorton committed:
Added Content-Type to error handler so it correctly encodes the not-quite-a-hyphen character.
Placing it after the engine.process() appears to rewrite with an ISO Content-Type.
Before appears to work, but I've have odd occurances with it sometimes being correct.
